test/Microsoft.SqlTools.ServiceLayer.TestDriver/Tests/QueryExecutionTests.cs

```diff
@@ -323,5 +324,38 @@ public class QueryExecutionTests
                 await testHelper.Disconnect(queryTempFile.FilePath);
             }
         }
+
+        [Fact]
+        public async Task NoOpQueryReturnsMessage()
```

Comment:
Yeah, that would make sense. After looking further into the code it appears that the message field is actually unused (aside from this use case for the bug) which I'm guessing is an artifact of changing query execution to be asynchronous and moving the messages to a different event. I had originally thought that a test should have already existed, but looks like we'll need one for the error message case too.